### PR TITLE
Adding bash Completions

### PR DESCRIPTION
**Pull request by rhatdan**
These are not ready to merge, mainly because I want to discuss the CLI.  I went and did a pass at this to see how buildah worked.  I think we need to improve some of the CLI before we announce this to the world.

### CODE REVIEW

**Comment by rhatdan** [comment]
Since the CLI has changed a lot, I think this patch could be looked at for merging.

File: contrib/completions/bash/buildah [hunk 1]
```diff
@@ -0,0 +1,461 @@
...
+
+# Subcommand processing.
+# Locates the first occurrence of any of the subcommands contained in the
+# first argument. In case of a match, calls the corresponding completion
+# function and returns 0.
+# If no match is found, 1 is returned. The calling function can then
+# continue processing its completion.
+#
+# TODO if the preceding command has options that accept arguments and an
+# argument is equal ot one of the subcommands, this is falsely detected as
```
Comment (TomSweeneyRedHat) [inline contrib/completions/bash/buildah]: nit, 'equal ot" to "equal to"

File: contrib/completions/bash/buildah [hunk 2]
```diff
@@ -0,0 +1,461 @@
...
+         if [ $cword -eq $counter ]; then
+             COMPREPLY=($(compgen -W "${commands[*]} help" -- "$cur"))
+         fi
+         ;;
+ esac
+}
+
+ _buildah_delete() {
+     local boolean_options="
+     --help
```
Comment (TomSweeneyRedHat) [inline contrib/completions/bash/buildah]: do we need to include debug here and in the other commands?

File: contrib/completions/bash/buildah (hunk 1 above)
Comment (rhatdan) [inline contrib/completions/bash/buildah]: Fixed

File: contrib/completions/bash/buildah (hunk 2 above)
Comment (rhatdan) [inline contrib/completions/bash/buildah]: No these are options of the `buildah` command, not buildah delete.

IE 

buildah -<tab> Shows them

buildah delete -<tab> shows these.

**Comment by rhatdan** [comment]
Ok these seem to work fairly well now.  I think it is time to merge.

**Comment by rhatdan** [comment]
@rh-atomic-bot r+ 4196e37

**Pull request closed without merging**